tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/SerilogTests.cs

```diff
@@ -35,22 +36,94 @@ public SerilogTests(ITestOutputHelper output)
             SetServiceVersion("1.0.0");
         }
 
-        // exclude loadFromConfig from v1.x as it's not available
-        public static IEnumerable<object[]> GetTestData()
-            => from packageVersion in PackageVersions.Serilog.SelectMany(x => x).Select(x => (string)x)
-               from enableLogShipping in new[] { true, false }
-               from enable128BitInjection in new[] { true, false }
-               from loadFromConfig in new[] { true, false }
-               where !loadFromConfig // only include loadFromConfig when >= 2.12.0 (early versions of the config package are buggy)
-                  || (!string.IsNullOrEmpty(packageVersion) && new Version(packageVersion) >= new Version("2.12.0"))
-               select new object[] { packageVersion, enableLogShipping, loadFromConfig, enable128BitInjection };
+        [SkippableTheory]
+        [CombinatorialOrPairwiseData]
+        [Trait("Category", "EndToEnd")]
+        [Trait("RunOnWindows", "True")]
+        [Trait("SupportsInstrumentationVerification", "True")]
+        public async Task InjectsLogsWhenEnabled(
+            [PackageVersionData(nameof(PackageVersions.Serilog), minInclusive: "2.12.0")] string packageVersion,
+            bool enableLogShipping,
+            bool loadFromConfig,
+            bool enable128BitInjection)
+        {
+            // only include loadFromConfig when >= 2.12.0 (early versions of the config package are buggy)
+            Skip.If(string.IsNullOrEmpty(packageVersion) && !EnvironmentHelper.IsCoreClr(), "Default version of Serilog for .NET Framework sample doesn't support load from config.");
```

Comment:
Nice, took me a while to get my head around all these, but I think this is the best we can do 👍

Reply:
Yeah...

I think the only way to make this cleaner would be to not return empty `packageVersion` when running locally or to implement some better way to exclude possible combinations, but that has proven difficult